pandas/tests/io/test_pickle.py

```diff
@@ -155,28 +157,44 @@ def test_pickles(current_pickle_data, legacy_pickle):
         compare(current_pickle_data, legacy_pickle, version)
 
 
-def test_round_trip_current(current_pickle_data):
-    def python_pickler(obj, path):
-        with open(path, "wb") as fh:
-            pickle.dump(obj, fh, protocol=-1)
+def python_pickler(obj, path):
+    with open(path, "wb") as fh:
+        pickle.dump(obj, fh, protocol=-1)
 
-    def python_unpickler(path):
-        with open(path, "rb") as fh:
-            fh.seek(0)
-            return pickle.load(fh)
 
+def python_unpickler(path):
+    with open(path, "rb") as fh:
+        fh.seek(0)
+        return pickle.load(fh)
+
+
+named_pickle_writers = {
+    "python": python_pickler,
+    "pandas_proto_default": pd.to_pickle,
+    "pandas_proto_highest": functools.partial(
+        pd.to_pickle, protocol=pickle.HIGHEST_PROTOCOL
+    ),
+    "pandas_proto_4": functools.partial(pd.to_pickle, protocol=4),
+}
+if sys.version_info >= (3, 8):
+    named_pickle_writers.update(
+        {
+            "pandas_proto_5": functools.partial(pd.to_pickle, protocol=5),
+        }
+    )
+pickle_writer_ids, pickle_writers = zip(*named_pickle_writers.items())
```

Comment:
alternativey you can use something like: 

```pytest.param(...., marks=pytest.mark.skipif(not PY38, reason=...)```

Reply:
`pytest.parm` is more readable - however, again do we want to skip for 3.9+?